src/iuse_software_lightson.cpp

```diff
@@ -0,0 +1,184 @@
+
+    werase( w );
+    draw_border( w );
+}
+
+void lightson_game::draw_level()
+{
+    for( int i = 0; i < level_size.first; i++ ) {
+        for( int j = 0; j < level_size.second; j++ ) {
+            bool selected = ( position.first == i && position.second == j );
+            bool on = level[i * level_size.second + j];
+            const nc_color fg = on ? c_white : c_dark_gray;
+            const char symbol = on ? '#' : '-';
+            mvwputch( w, i + 1, j + 1, selected ? hilite( c_white ) : fg, symbol );
+        }
+    }
+    wrefresh( w );
+}
+
+void lightson_game::generate_change_coords( int changes )
+{
+    change_coords.resize( changes );
+    const int size = level_size.first * level_size.second;
+
+    std::pair< int, int > candidate;
+    for( int k = 0; k < changes; k++ ) {
+        do {
+            const int candidate_index = rng( 0, size - 1 );
+
+            const int col = candidate_index % level_size.second;
+            const int row = ( candidate_index - col ) / level_size.second;
+            candidate = std::make_pair( row, col );
+            // not accept repeatable coordinates
+        } while( !( k == 0 ||
+                    std::find( change_coords.begin(), change_coords.end(), candidate ) == change_coords.end() ) );
+        change_coords[k] = candidate;
+    }
+}
+
+bool lightson_game::check_win()
+{
+    return std::all_of( level.begin(), level.end(), []( bool i ) {
+        return i;
+    } );
+}
+
+void lightson_game::toggle_lights()
+{
+    const int row = position.first;
+    const int col = position.second;
+    const int height = level_size.first;
+    const int width = level_size.second;
+
+    if( row > 0 ) {
+        level[( row - 1 ) * width + col] = !level[( row - 1 ) * width + col]; // N
+    }
+    level[row * width + col] = !level[row * width + col]; // x
+    if( row < height - 1 ) {
+        level[( row + 1 ) * width + col] = !level[( row + 1 ) * width + col]; // S
+    }
+
+    if( col > 0 ) {
+        level[row * width + ( col - 1 )] = !level[row * width + ( col - 1 )]; // W
+    }
+    if( col < width - 1 ) {
+        level[row * width + ( col + 1 )] = !level[row * width + ( col + 1 )]; // E
+    }
+}
+
+int lightson_game::start_game()
+{
+    const int w_height = 15;
+    const int iOffsetX = ( TERMX > FULL_SCREEN_WIDTH ) ? ( TERMX - FULL_SCREEN_WIDTH ) / 2 : 0;
+    const int iOffsetY = ( TERMY > FULL_SCREEN_HEIGHT ) ? ( TERMY - FULL_SCREEN_HEIGHT ) / 2 : 0;
+    w_border = catacurses::newwin( w_height, FULL_SCREEN_WIDTH, iOffsetY, iOffsetX );
+    w = catacurses::newwin( w_height - 6, FULL_SCREEN_WIDTH - 2, iOffsetY + 1, iOffsetX + 1 );
+    draw_border( w_border );
+
+    input_context ctxt( "LIGHTSON" );
+    ctxt.register_directions();
+    ctxt.register_action( "TOGGLE" );
+    ctxt.register_action( "RESET" );
+    ctxt.register_action( "QUIT" );
+
+    std::vector<std::string> shortcuts;
+    shortcuts.push_back( _( "<5>toggle lights" ) ); // '5': toggle
```

Comment:
Maybe add space after `<5>`? Also, capitalization? Also, maybe change shortcut to spacebar?